### PR TITLE
Override target port used when creating openshift route

### PR DESCRIPTION
**Pull request by Dairdevil**
Addressing issue https://github.com/quarkusio/quarkus/issues/26587

WIP, wanted to check with @geoand and/or @iocanel if you're happy with this approach? This change should have broken one of the ITs which I will update and extend if you think I'm heading in the right direction with this

### CODE REVIEW

**Comment by geoand** [comment]
Thanks for this!

Seems reasonable to me, but I will leave it up to @iocanel to decide

**Comment by Dairdevil** [comment]
Updated the existing IT and added a new one to include covering the new config option

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessorUtil.java
Comment (gastaldi) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessorUtil.java:12]: Just as in https://github.com/quarkusio/quarkus/pull/27027, it would be nice to have this in a `@ConfigItem`

Comment (Dairdevil) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/OpenshiftProcessorUtil.java:12]: I’ve made this change but also had a bit of a nightmare with git which has meant some of these changes shouldn’t still be here. Will resolve in the morning!

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java
Comment (Dairdevil) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java:20]: I don't like coding by exception but struggled to think of an alternative here to ensuring that numerical values are treated properly. Without this, if the value in the property is set to a number it is written to the yml file quoted as a string regardless. I believe this will have no functional impact, but would still represent an undesirable change to how it currently works. Any thoughts/suggestions on how to approach this?

Comment (gastaldi) [reply extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java:20]: I don't see any no problem in having it written as a String all the time

File: integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithRoutePropertiesTest.java
Comment (iocanel) [inline integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithRoutePropertiesTest.java:66]: this looks weird

Comment (iocanel) [reply integration-tests/kubernetes/quarkus-standard-way/src/test/java/io/quarkus/it/kubernetes/OpenshiftWithRoutePropertiesTest.java:66]: this won't owrk

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/ApplyRouteTargetPortConfigurator.java:9]: We use Configurator for other things this is a Decorator and should be named as such

File: extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RouteConfig.java
Comment (iocanel) [inline extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/RouteConfig.java:31]: This is misleading. I think the rest of the code code doesn't handle int values (it always treats it as string)

**Review by iocanel — CHANGES_REQUESTED** [review]
Thanks for the pr.
It has some minor issues that need to be addressed.

**Comment by iocanel** [comment]
Also, I'd like to mention that this also needs to be adressed on dekorate (which is currently in progress), so in the future this will be redundant. Till then, I would glaldy accept this PR once the minor changes are addressed.

**Comment by Dairdevil** [comment]
Thanks for the review @iocanel, making changes now

**Review by gsmet — DISMISSED** [review]
A reminder that we need to get rid of the merge commit before merging.

**Comment by Dairdevil** [comment]
Thanks @gsmet just doing that now. I might be having the same issue as my other PR where the CI checks didn't trigger, am I doing something wrong?

**Comment by geoand** [comment]
You aren't doing anything wrong :).

It's just the default setting kicking on.

**Comment by Dairdevil** [comment]
Apologies, I have had an absolute nightmare with git here. Ended up doing a fresh branch from main and reapplying the changes so hopefully its all clean now, however the bot has been confused as a result of my messing about and applied a load of tags.

**Comment by geoand** [comment]
No problem at all! Thanks for persevering!

**Review by iocanel — APPROVED** [review]
My concerns have been addressed, so it looks good.

**Review by iocanel — CHANGES_REQUESTED** [review]
Just noticed the ci issues.

**Comment by iocanel** [comment]
@Sgitario: AFAIR, you were working on this from the dekorate side. What is the current status? Is this still relevant?

**Comment by Sgitario** [comment]
> @Sgitario: AFAIR, you were working on this from the dekorate side. What is the current status? Is this still relevant?

Indeed, this is already supported on dekorate side: https://github.com/dekorateio/dekorate/commit/ef3321e4fab7496175a17066fda75602cd6af3e4.
However, there are some necessary changes on Quarkus side too. I will address this in a separated pull request. 
Thanks for the ping!

**Comment by Sgitario** [comment]
Closing this pull request as the changes became obsolete after bumping Dekorate to the latest version.
The new pull request is https://github.com/quarkusio/quarkus/pull/29120.

**Pull request closed without merging**
